source/slang/lower-to-ir.cpp

```diff
@@ -3011,6 +3040,19 @@ LoweredValInfo tryGetAddress(
             // as part of the subscript operation being referenced.
             //
             auto subscriptInfo = val.getBoundSubscriptInfo();
+
+            // We don't want to immediately bind to a `ref` accessor if there is
+            // a `set` accessor available, unless we are in an "aggressive" mode
+            // where we really want/need a pointer to be able to make progress.
+            //
+            if(mode != TryGetAddressMode::Aggressive
+                && getMembersOfType<SetterDecl>(subscriptInfo->declRef).Count())
```

Comment:
Guess might be better to have a 'hasMembersOfType'

Reply:
Yes, that would be a good convenience function to provide.